pkg/config/config.go

```diff
@@ -31,3 +39,32 @@ func NewDefault() *Config {
 type Version struct {
 	Format string `mapstructure:"format"`
 }
+
+func IsGreaterThanOrEqualGo118(v string) bool {
+	v1, err := hcversion.NewVersion(strings.TrimPrefix(v, "go"))
+	if err != nil {
+		return false
+	}
+
+	limit, err := hcversion.NewVersion("1.18")
+	if err != nil {
+		return false
+	}
+
+	return v1.GreaterThanOrEqual(limit)
+}
+
+func DetectGoVersion() string {
```

Comment:
Is this function detecting the `version` directive in the `go.mod` file or the installed version of `Go`?

Reply:
The explanation is in the PR description, and the example.golangci.yml